pkg/operator/ceph/object/objectstore.go

```diff
@@ -749,12 +752,22 @@ func CreatePools(context *Context, clusterSpec *cephv1.ClusterSpec, metadataPool
 		logger.Warningf("failed to adjust the PG count for rgw metadata pools. using the general default. %v", err)
 		metadataPoolPGs = cephclient.DefaultPGCount
 	}
+	if metadataPool.AutoscaleMode == "" {
+		metadataPool.AutoscaleMode = cephclient.DefaultAutoscaleMode
+	}
 
 	if err := createSimilarPools(context, append(metadataPools, rootPool), clusterSpec, metadataPool, metadataPoolPGs); err != nil {
 		return errors.Wrap(err, "failed to create metadata pools")
 	}
-
-	if err := createSimilarPools(context, []string{dataPoolName}, clusterSpec, dataPool, cephclient.DefaultPGCount); err != nil {
+	if dataPool.PGs > 0 {
```

Comment:
Lines 762-769 look the same as in pool.go. How about refactoring to a helper method to avoid duplication?